src/transformations/vertoffset.cpp

```diff
@@ -0,0 +1,90 @@
+/************************************************************************
+* Copyright (c) 2022, Even Rouault <even.rouault at spatialys.com>
+*
+* Permission is hereby granted, free of charge, to any person obtaining a
+* copy of this software and associated documentation files (the "Software"),
+* to deal in the Software without restriction, including without limitation
+* the rights to use, copy, modify, merge, publish, distribute, sublicense,
+* and/or sell copies of the Software, and to permit persons to whom the
+* Software is furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included
+* in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+* OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+* THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+* DEALINGS IN THE SOFTWARE.
+*
+***********************************************************************/
+#define PJ_LIB__
+
+#include <errno.h>
+#include <math.h>
+
+#include "proj.h"
+#include "proj_internal.h"
+
+PROJ_HEAD(vertoffset, "Vertical Offset and Slope");
+
+namespace { // anonymous namespace
+struct pj_opaque_vertoffset {
+    double slope_lon;
+    double slope_lat;
+    double zoff;
+    double rho0;
+    double nu0;
+};
+} // anonymous namespace
+
+// Cf EPSG Dataset coordinate operation method code 1046 "Vertical Offset and Slope"
+
+static PJ_XYZ forward_3d(PJ_LPZ lpz, PJ *P) {
+    const struct pj_opaque_vertoffset *Q = (const struct pj_opaque_vertoffset *) P->opaque;
+    PJ_XYZ xyz;
+    xyz.x = lpz.lam + P->lam0;
```

Comment:
Since this is a bit unusual a comment explaining what's going on would be nice

Reply:
comment added